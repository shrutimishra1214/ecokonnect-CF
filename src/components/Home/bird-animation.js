import React from 'react';
import styled, { keyframes } from 'styled-components';

const flyCycle = keyframes`
	100% {background-position: -900px 0}
`;

const flyRightOne = keyframes`
		0% {transform: scale(0.3) translateX(-10vw)}
		10% {transform: translateY(2vh) translateX(10vw) scale(0.4)}
		20% {transform: translateY(0vh) translateX(30vw) scale(0.5)}
		30% {transform: translateY(4vh) translateX(50vw) scale(0.6)}
		40% {transform: translateY(2vh) translateX(70vw) scale(0.6)}
		50% {transform: translateY(0vh) translateX(90vw) scale(0.6)}
		60% {transform: translateY(0vh) translateX(110vw) scale(0.6)}
		100% {transform: translateY(0vh) translateX(110vw) scale(0.6)}
`;

const Container = styled.div`
	z-index: -1;
	position: absolute;
	top: 0;
	left: 0;
	overflow: hidden;
	width: 100vw;
	background-color: transparent;
	height: 100vh;
	@media only screen and (max-width: 800px) {
		height: 30vh;
	}
`;

const Bird = styled.div`
	background-image: url(https://s3-us-west-2.amazonaws.com/s.cdpn.io/174479/bird-cells-new.svg);
	background-size: auto 100%;
	width: 88px;
	height: 125px;
	will-change: background-position;
	animation-name: ${flyCycle};
	animation-timing-function: steps(10);
	animation-iteration-count: infinite;
`;

const BirdOne = styled(Bird)`
	animation-duration: 1s;
	animation-delay: -0.5s;
`;
const BirdTwo = styled(Bird)`
	animation-duration: 1.25s;
	animation-delay: -0.25s;
`;

const BirdContainer = styled.div`
	position: absolute;
	top: 20%;
	left: -10%;
	transform: scale(0) translateX(-10vw);
	will-change: transform;
	animation-name: ${flyRightOne};
	animation-timing-function: linear;
	animation-iteration-count: infinite;
`;

const BirdContainerOne = styled(BirdContainer)`
	animation-duration: 15s;
	animation-delay: 0;
`;

const BirdContainerTwo = styled(BirdContainer)`
	animation-duration: 14.6s;
	animation-delay: 9.5s;
`;

const BirdAnimation = () => {
	return (
		<Container>
			<BirdContainerOne>
				<BirdOne></BirdOne>
			</BirdContainerOne>
			<BirdContainerTwo>
				<BirdTwo></BirdTwo>
			</BirdContainerTwo>
		</Container>
	);
};

export default BirdAnimation;
