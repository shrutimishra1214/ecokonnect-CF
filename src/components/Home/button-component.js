import React from 'react';
import styled from 'styled-components';

const Button = styled.button`
	background: transparent;
	backdrop-filter: blur(2px);
	padding: 8px;
	font-size: 20px;
	border: 2px solid;
	border-radius: 5px;
	font-weight: 500;
	text-transform: uppercase;
	transition: background-color 1s ease;
	color: white;
	border-color: white;
	background: forestgreen;
	z-index: 0;
	transition: all 300ms;
	&:hover {
		cursor: pointer;
		transform: scale(1.1);
	}
	@media only screen and (max-width: 950px) {
		width: 14em;
		padding: 0.6em;
		margin: 0.3em;
		font-size: 16px;
	}
	@media only screen and (max-width: 500px) {
		font-size: 15px;
	}
`;

const ButtonComponent = ({ content }) => {
	return <Button>{content}</Button>;
};

export default ButtonComponent;
