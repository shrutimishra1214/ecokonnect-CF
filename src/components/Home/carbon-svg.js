import React from 'react';
import styled, { keyframes } from 'styled-components';

const dash = keyframes`
	to{stroke-dashoffset: 0}
`;

const SVG = styled.svg`
	-webkit-filter: drop-shadow(3px 3px 2px rgba(0, 0, 0, 0.7));
	filter: drop-shadow(3px 3px 2px rgba(0, 0, 0, 0.7));
	/* Similar syntax to box-shadow */
`;

const Path = styled.path`
	stroke-dasharray: 500;
	stroke-dashoffset: 500;
	stroke-width: 1;
	animation: ${dash} 10s linear forwards;
`;

const PathLeter = styled.path`
	stroke: green;
	stroke-dasharray: 100;
	stroke-dashoffset: 100;
	stroke-width: 0.5;
	animation: ${dash} 5s linear forwards;
`;

const IntroSVG = () => {
	return (
		<SVG xmlns='http://www.w3.org/2000/svg' id='Icons' viewBox='0 0 74 74' width='100' height='100'>
			<Path
				fill='transparent'
				stroke='#000000'
				d='M61.347,58.512H12.653a10.653,10.653,0,0,1-.969-21.262A14.288,14.288,0,0,1,30.562,24.7,17.272,17.272,0,0,1,62.518,37.271a10.653,10.653,0,0,1-1.171,21.242ZM25.926,25.934A12.287,12.287,0,0,0,13.653,38.207a1,1,0,0,1-1,1,8.653,8.653,0,1,0,0,17.306H61.347a8.653,8.653,0,1,0,0-17.306h-.164a1,1,0,0,1-.944-1.328,15.536,15.536,0,0,0,.886-5.1,15.281,15.281,0,0,0-29.162-6.4,1,1,0,0,1-1.295.5A12.192,12.192,0,0,0,25.926,25.934Z'
			/>
			<PathLeter d='M27.95,50.682a6.29,6.29,0,0,1-3.468-.871,5.461,5.461,0,0,1-2.039-2.521,8.953,8.953,0,0,1-.693-3.514,7.78,7.78,0,0,1,.785-3.58,5.443,5.443,0,0,1,2.246-2.336,6.568,6.568,0,0,1,3.2-.8,5.446,5.446,0,0,1,3.338,1.013A5.271,5.271,0,0,1,33.2,40.92l.037.126-1.926.453-.035-.11A3.89,3.89,0,0,0,30.018,39.4a3.37,3.37,0,0,0-2.076-.619,4.08,4.08,0,0,0-2.419.69,3.671,3.671,0,0,0-1.36,1.856,7.623,7.623,0,0,0-.4,2.442,7.816,7.816,0,0,0,.472,2.833,3.471,3.471,0,0,0,1.442,1.776,4.08,4.08,0,0,0,2.131.591,3.572,3.572,0,0,0,2.354-.8,4.138,4.138,0,0,0,1.308-2.39l.027-.126,1.957.494-.031.121a5.9,5.9,0,0,1-1.979,3.282A5.48,5.48,0,0,1,27.95,50.682Zm.026-13.367a6.321,6.321,0,0,0-3.075.765,5.19,5.19,0,0,0-2.144,2.231A7.524,7.524,0,0,0,22,43.776a8.7,8.7,0,0,0,.674,3.417A5.205,5.205,0,0,0,24.619,49.6a6.045,6.045,0,0,0,3.331.831,5.239,5.239,0,0,0,3.34-1.08,5.6,5.6,0,0,0,1.858-3.026l-1.463-.37a4.311,4.311,0,0,1-1.363,2.4,3.82,3.82,0,0,1-2.514.857,4.332,4.332,0,0,1-2.26-.626A3.713,3.713,0,0,1,24,46.691a8.032,8.032,0,0,1-.489-2.924,7.874,7.874,0,0,1,.412-2.522,3.909,3.909,0,0,1,1.452-1.979,4.317,4.317,0,0,1,2.563-.737,3.617,3.617,0,0,1,2.227.668A4.055,4.055,0,0,1,31.48,41.2l1.443-.34a4.985,4.985,0,0,0-1.76-2.586A5.2,5.2,0,0,0,27.977,37.314Z' />
			<PathLeter d='M41.07,50.682a6.262,6.262,0,0,1-3.328-.9,5.824,5.824,0,0,1-2.225-2.459,7.439,7.439,0,0,1-.753-3.282,7.177,7.177,0,0,1,1.762-5.125,6.022,6.022,0,0,1,4.553-1.86,6.344,6.344,0,0,1,3.291.873,5.723,5.723,0,0,1,2.239,2.437,7.9,7.9,0,0,1,.767,3.525,7.784,7.784,0,0,1-.807,3.575,5.57,5.57,0,0,1-2.3,2.4A6.575,6.575,0,0,1,41.07,50.682Zm.009-13.376a5.781,5.781,0,0,0-4.371,1.782,6.937,6.937,0,0,0-1.693,4.953,7.2,7.2,0,0,0,.728,3.172,5.581,5.581,0,0,0,2.131,2.355,6.016,6.016,0,0,0,3.2.863,6.329,6.329,0,0,0,3.078-.782,5.325,5.325,0,0,0,2.2-2.3,7.528,7.528,0,0,0,.779-3.461,7.65,7.65,0,0,0-.741-3.415,5.487,5.487,0,0,0-2.142-2.333A6.1,6.1,0,0,0,41.079,37.306Zm-.018,11.912a4.274,4.274,0,0,1-3.244-1.384,5.325,5.325,0,0,1-1.29-3.767c0-1.972.454-3.4,1.349-4.256a4.511,4.511,0,0,1,3.212-1.282,4.368,4.368,0,0,1,2.372.667,4.24,4.24,0,0,1,1.608,1.879,6.792,6.792,0,0,1,.544,2.807,5.631,5.631,0,0,1-1.28,3.938A4.265,4.265,0,0,1,41.062,49.218Zm.026-10.439a4.272,4.272,0,0,0-3.04,1.213c-.843.8-1.271,2.174-1.271,4.076A5.076,5.076,0,0,0,38,47.663a4.258,4.258,0,0,0,6.147-.012,5.4,5.4,0,0,0,1.214-3.769,6.526,6.526,0,0,0-.522-2.7,4,4,0,0,0-1.513-1.771A4.127,4.127,0,0,0,41.088,38.778Z' />
			<path
				stroke='transparent'
				d='M55.563,51.24h-5.3L50.252,51a1.835,1.835,0,0,1,.121-.718A3.518,3.518,0,0,1,51,49.24a9.446,9.446,0,0,1,1.2-1.134,9.71,9.71,0,0,0,1.56-1.5,1.613,1.613,0,0,0,.369-.927.942.942,0,0,0-.3-.715,1.118,1.118,0,0,0-.817-.3,1.145,1.145,0,0,0-.855.314,1.213,1.213,0,0,0-.319.9l0,.275L50.4,46.01l.022-.246a2.506,2.506,0,0,1,.795-1.731,2.669,2.669,0,0,1,1.817-.6,2.557,2.557,0,0,1,1.83.65,2.158,2.158,0,0,1,.688,1.621,2.44,2.44,0,0,1-.2.956,3.607,3.607,0,0,1-.637.958,14.729,14.729,0,0,1-1.4,1.306c-.525.44-.866.742-1.013.9l-.058.062h3.318Zm-4.8-.5h4.3v-.359H51.3l.23-.38a3.143,3.143,0,0,1,.409-.526c.162-.171.507-.477,1.055-.936a14.4,14.4,0,0,0,1.351-1.255,3.106,3.106,0,0,0,.55-.82,1.946,1.946,0,0,0,.159-.761,1.651,1.651,0,0,0-.533-1.259,2.048,2.048,0,0,0-1.485-.512,2.191,2.191,0,0,0-1.488.477,1.873,1.873,0,0,0-.595,1.156l.4.042a1.613,1.613,0,0,1,.449-.974,1.641,1.641,0,0,1,1.209-.46,1.616,1.616,0,0,1,1.164.435,1.438,1.438,0,0,1,.456,1.076,2.088,2.088,0,0,1-.466,1.223,9.964,9.964,0,0,1-1.646,1.587,9.01,9.01,0,0,0-1.137,1.071,3.043,3.043,0,0,0-.544.894A1.387,1.387,0,0,0,50.766,50.74Z'
			/>
		</SVG>
	);
};

export default IntroSVG;
