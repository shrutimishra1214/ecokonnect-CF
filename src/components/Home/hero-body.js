import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

import ButtonComponent from './button-component';
import HeroText from './hero-text';

const ButtonBox = styled.div`
	display: flex;
	align-items: center;
	justify-content: space-around;
	margin: auto;
	padding-top: 2em;
	${({ theme }) =>
		theme.query.desktop({
			display: 'grid',
			'align-items': 'center',
			'justify-content': 'center',
			'padding-top': '0',
		})}
`;

const HeroBody = () => {
	return (
		<>
			<HeroText>Select the type of carbon footprint calculation</HeroText>
			<ButtonBox>
				<Link to='/quiz/person'>
					<ButtonComponent content={'Individual'} />
				</Link>
				<Link to='/quiz/organization'>
					<ButtonComponent content={'Small Organization'} />
				</Link>
			</ButtonBox>
		</>
	);
};

export default HeroBody;
