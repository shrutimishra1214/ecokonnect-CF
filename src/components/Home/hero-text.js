import React from 'react';
import styled from 'styled-components';

const TextStyle = styled.h2`
	text-align: center;
	color: ${(props) => props.theme.colors.secondary};
	font-weight: 700;
	font-size: 20px;
	padding-top: 2em;
	margin-bottom: 0;
	${({ theme }) =>
		theme.query.desktop({
			'font-size': '18px',
			'margin-bottom': '1em',
		})}
	${({ theme }) =>
		theme.query.bigMobile({
			'font-size': '15px',
			'margin-bottom': '1em',
		})}
	${({ theme }) =>
		theme.query.smallMobile({
			'font-size': '12px',
			'margin-bottom': '1em',
		})}
`;

const HeroText = ({ children }) => {
	return <TextStyle>{children}</TextStyle>;
};

export default HeroText;
