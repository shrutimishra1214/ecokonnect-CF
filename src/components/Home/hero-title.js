import React from 'react';
import styled from 'styled-components';

const Title = styled.h1`
	text-transform: uppercase;
	color: ${(props) => props.theme.colors.forest};
	font-family: ${(props) => props.theme.fontFamily.title};
	font-weight: 900;
	text-align: center;
	font-size: 80px;
	margin-bottom: 0;
	letter-spacing: -5px;
	${({ theme }) =>
		theme.query.desktop({
			'font-size': '75px',
		})}
	${({ theme }) =>
		theme.query.smallDesktop({
			'margin-top': '0.3em',
		})}
	${({ theme }) =>
		theme.query.bigMobile({
			'font-size': '60px',
		})}
	${({ theme }) =>
		theme.query.smallMobile({
			'font-size': '55px',
		})}
`;

const Subtitle = styled.h4`
	text-transform: uppercase;
	color: ${(props) => props.theme.colors.forest};
	font-weight: 500;
	text-align: center;
	font-size: 20px;
	margin-top: 0.3em;
	${({ theme }) =>
		theme.query.desktop({
			'font-size': '16px',
		})}
	${({ theme }) =>
		theme.query.bigMobile({
			'font-size': '14px',
		})}
	${({ theme }) =>
		theme.query.smallMobile({
			'font-size': '12px',
		})}
`;

const HeroTitle = ({ children }) => {
	return (
		<>
			<Title>{children}</Title>
			<Subtitle>Are you being all the green that you can be?</Subtitle>
		</>
	);
};

export default HeroTitle;
