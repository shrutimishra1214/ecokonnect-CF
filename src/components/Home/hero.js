import React from 'react';
import HeroTitle from './hero-title';
import HeroBody from './hero-body';

const Hero = () => {
	return (
		<>
			<HeroTitle>Carbon Footprint</HeroTitle>
			<HeroBody />
		</>
	);
};

export default Hero;
