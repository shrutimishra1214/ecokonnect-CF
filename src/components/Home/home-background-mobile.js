import React from 'react';
import styled from 'styled-components';
import tree from '../../assets/bg_home.svg';
import BirdAnimation from './bird-animation';

const BgImg = styled.img`
	width: 30em;
	padding-left: 2em;
	@media only screen and (max-width: 1000px) {
		width: 24em;
		padding-left: 2em;
	}
	@media only screen and (max-width: 800px) {
		width: 14em;
		padding-left: 0em;
	}
	@media only screen and (max-width: 500px) {
		width: 12em;
		padding-left: 0em;
	}
	@media only screen and (max-width: 350px) {
		width: 8em;
	}
`;

const MobileHomeBackground = () => {
	return (
		<>
			<BgImg src={tree} alt='tree' />
			<BirdAnimation />
		</>
	);
};

export default MobileHomeBackground;
