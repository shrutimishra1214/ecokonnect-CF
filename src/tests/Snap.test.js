import React from 'react';
import ReactDOM from 'react-dom';
import Title from '../components/Home/intro-title';
import Home from '../components/Home/home-page';
import Theme from '../Theme';

import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('renders correctly enzyme', () => {
	const wrapper = shallow(
		<Theme>
			<Title />
		</Theme>
	);

	expect(toJson(wrapper)).toMatchSnapshot();
});

// it('renders correctly enzyme', () => {
// 	const wrapper = mount(
// 		<Theme>
// 			<Home />
// 		</Theme>
// 	);

// 	wrapper.find('button').at(0).simulate('click');

// 	const counterWrapper = wrapper.find(Title);
// 	expect(counterWrapper.find('h2')).toBe('1');
// });
