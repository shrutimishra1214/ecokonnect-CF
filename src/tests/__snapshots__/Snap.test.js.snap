// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly enzyme 1`] = `
<ThemeProvider
  theme={
    Object {
      "breakpoints": Object {
        "l": 800,
        "m": 550,
        "s": 450,
        "xl": 950,
        "xs": 350,
        "xxl": 1200,
      },
      "colors": Object {
        "forest": "#38a66d",
        "lightForest": "#03b554",
        "primary": "white",
        "secondary": "black",
      },
      "fontFamily": Object {
        "title": "Yanone Kaffeesatz, sans-serif",
        "title2": "Raleway, sans-serif",
      },
      "fontSizes": Object {
        "header": "16",
        "subheader": "16",
        "subtitle": "16",
        "text": "16",
        "title": "16",
      },
      "fontWeights": Object {
        "bold": "700",
        "light": "400",
        "lighter": "300",
        "normal": "500",
      },
      "query": Object {
        "bigDesktop": [Function],
        "bigMobile": [Function],
        "desktop": [Function],
        "mobile": [Function],
        "smallDesktop": [Function],
        "smallMobile": [Function],
      },
    }
  }
>
  <Title />
</ThemeProvider>
`;
